maestro-android/src/androidTest/java/dev/mobile/maestro/MaestroDriverService.kt

```diff
@@ -76,7 +85,11 @@ class MaestroDriverService {
         val uiDevice = UiDevice.getInstance(instrumentation)
         val uiAutomation = instrumentation.uiAutomation
 
-        NettyServerBuilder.forPort(7001)
+        val port = InstrumentationRegistry.getArguments().getString("port", "9001").toInt()
+
+        println("Server running on port [ $port ]")
```

Comment:
let me explain - the number 7001 or 9001 really doesn't matter any more

this is actually now set by the parameter which gets passed down from default CLI parameters, unless it's specified. A random port between 9001 and 9128 is selected at runtime. 

Below addition from App.kt ;
```
@Option(names = ["--driver-host-port"], hidden = true)
    var driverHostPort: Int = Random.nextInt(9001, 9128)
```

This allows multiple maestro executables to run simultaneously. It's also a required key part of bringing up more than one device. 

Why 128 ? That is because adb allows for up to 64 server / serial port pairs to be set for one instance of adb server that drives these emulators. 

Why 9001 ? It's just a fallback number, it's always going to be set by the initial random number that is going to be above 9001 and below 9128. 

( changing this back to 7001 is not going to change anything, the driver host port parameter is now always used, but I left the fallback port in case studio or something else still uses driver launch without the host port ? )

Reply:
Yeah, I totally get that. 

> It's user's responsibility to make sure their ports are usable

Agreed, however, imagine that the user already took their time to setup maestro over the network, and succeeded by making sure that 7001 is a usable port. This change will now break what the user thinks is an already solved problem. 

Therefore, I propose that if sharding / parallel execution **is not being used** then 7001 should still be the default driver port. This way we avoid unnecessarily breaking the contract without, of course, the actual need to.

Reply:
of course, including this in the next  👍